src/test/javascript/spec/component/course/detail/course-detail-bar-chart.component.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { RouterTestingModule } from '@angular/router/testing';
+import { TranslateService } from '@ngx-translate/core';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe.ts';
+import * as chai from 'chai';
+import { MockPipe } from 'ng-mocks';
+import { ChartsModule } from 'ng2-charts';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { of } from 'rxjs/internal/observable/of';
+import * as sinonChai from 'sinon-chai';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { CourseDetailBarChartComponent } from 'app/course/manage/detail/course-detail-bar-chart.component';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { MockSyncStorage } from '../../../helpers/mocks/service/mock-sync-storage.service';
+import { MockTranslateService } from '../../../helpers/mocks/service/mock-translate.service';
+import { ArtemisTestModule } from '../../../test.module';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('CourseDetailBarChartComponent', () => {
+    let fixture: ComponentFixture<CourseDetailBarChartComponent>;
+    let component: CourseDetailBarChartComponent;
+    let service: CourseManagementService;
+
+    const initialStats = [26, 46, 34, 12];
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([]), ChartsModule],
+            declarations: [CourseDetailBarChartComponent, MockPipe(ArtemisTranslatePipe)],
+            providers: [
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CourseDetailBarChartComponent);
+                component = fixture.componentInstance;
+                service = TestBed.inject(CourseManagementService);
+            });
+    });
+
+    beforeEach(() => {
+        component.courseId = 1;
+        component.numberOfStudentsInCourse = 50;
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should initialize', () => {
+        const graphData: number[] = [];
+        const spy = spyOn(service, 'getStatisticsData');
+        for (let i = 0; i < 4; i++) {
+            graphData[i] = 40 + 2 * i;
+        }
+        spy.and.returnValue(of(graphData));
```

Comment:
What happens to this, you are not using graphData later, not checking if the service method has been called?

Reply:
I use the `graphData` for the service call and i check the service call by checking the values in `component.data`. Since we initialize `data` in percentage and we said to have 50 people in the course, the expected value is double the `graphData` values